src/sentry/static/sentry/app/views/organizationDiscover/utils.jsx

```diff
@@ -0,0 +1,30 @@
+export function getQueryFromQueryString(queryString) {
+  if (!queryString) {
+    return {};
+  }
+  let parsedQuery = queryString;
+  let result = {};
+  parsedQuery = parsedQuery.replace(/^\?/, '').split('&');
+  parsedQuery.forEach(item => {
+    if (item.includes('=')) {
+      let key = item.split('=')[0];
+      let value = JSON.parse(decodeURIComponent(item.split('=')[1]));
+      result[key] = value;
+    }
+  });
+
+  return result;
+}
+
+export function getQueryStringFromQuery(query) {
+  let str = '?';
+
+  let keys = Object.keys(query);
+  keys.sort();
+  for (let i = 0; i < keys.length; ++i) {
+    console.log(keys[i], ': ', query[keys[i]]);
+    str += keys[i] + '=' + encodeURIComponent(JSON.stringify(query[keys[i]])) + '&';
```

Comment:
I think we'd need to encode the key here

Reply:
btw, you can also do

```
const queryProperties = Object.entries(query).map(([key, value]) => {
    return encodeURIComponent(key) + '=' + 
encodeURIComponent(JSON.stringify(value));
  });

  return `?${queryProperties.join('&')}`;
```
which has the benefit of not having the trailing ampersand

Reply:
So I looked it up and the trailing ampersand is actually completely legal and has no repercussions  -- I can still change it though.